### DISCUSSION TITLE
IOr value

### DISCUSSION THREAD

**Discussion by etsinko** [vavr-io/vavr#2867] [Ideas]
Hello,
I have been using Vavr extensively in my projects and had a need for an IOr value type. I understand that there is no such value type in Vavr and was wondering if Vavr needs it. In my opinion, it would be a nice addition for Vavr. 
I have already implemented it and would love to contribute it to the project. Let me know if you are interested in it and I'll create a PR.

**Comment by danieldietrich** [comment]
Hi Egor, thank you for your suggestion. Vavr aligns to the Scala standard library, therefore IOr would be a better fit for an extension library (on your side) on top of Vavr.

Just out of curiosity - is your IOr the bitwise Boolean inclusive-OR or the IO reader monad? Nevertheless, it would be an extension to Vavr.

Thanks!

**Comment by etsinko** [comment]
Hi Daniel, 
forgot to mention, this is inclusive-or which compliments either (exclusive-or) and it was inspired by Cats IOr from Typelevel. The use case would be similar to Either but in cases when you have non-fatal errors on the left side that you want to collect.
It is ok if you think that IOr wouldn't fit into Vavr because it is not the part of Scala standard library. However, Vavr's Validation is not part of the Scala standard library too but it fits nice in Vavr.

**Comment by danieldietrich** [comment]
Hi, thanks for the clarification!
Yes, you are right, Validation is fitting well into Vavr on the one hand but it is not part of the standard library on the other hand. My last efforts focused on simplifying Vavr. It would be good to move Validation out of Vavr, into its own library that builds on top of Vavr, similar like Cats built on top of Scala. IOr could then accompany Validation.

**Comment by etsinko** [comment]
Sounds good!
Let me know if there will be a project that builds on top of Vavr. I'd be happy to contribute.
